### PR TITLE
feat/x-support

### PR DESCRIPTION
**Pull request by wash2**
This adds optional support for X via wayland-proxy-virtwl.

### CODE REVIEW

File: data/wayland-proxy-virtwl.service [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[Unit]
+Description=Wayland-proxy-virtwl
+After=cosmic-session.target
+WantedBy=cosmic-session.target
+
+[Service]
+ExecStart=/var/bin/wayland-proxy-virtwl --wayland-display wayland-0 --x-display=0 --xrdb Xft.dpi:150 --x-unscale=2
```
Comment (Drakulix) [inline data/wayland-proxy-virtwl.service:7]: Do we want to hard code dpi and unscaling? I don't think so.

This is better handled by the user doing an override with `systemctl --user edit wayland-proxy-virtwl`.

File: justfile [hunk 2]
```diff
@@ -38,11 +41,17 @@ sysext dir=`echo $(pwd)/cosmic-sysext` version=("nightly-" + `git rev-parse --sh
...
     {{ make }} -C cosmic-panel install DESTDIR={{dir}} prefix=/usr
     {{ make }} -C cosmic-settings-daemon install DESTDIR={{dir}} prefix=/usr
-    {{ just }} rootdir={{dir}} cosmic-session/install
     {{ just }} _meson_install iced-workspaces-applet {{dir}}
     {{ just }} _meson_install user-color-editor {{dir}}
     {{ make }} -C xdg-desktop-portal-cosmic install DESTDIR={{dir}} prefix=/usr
-
+    if test {{x}} = 1; then
+        mkdir -p $HOME/.local/share/systemd/user/
+        install -Dm0644 data/wayland-proxy-virtwl.service $HOME/.local/share/systemd/user/wayland-proxy-virtwl.service
```
Comment (Drakulix) [inline justfile:49]: Lets install to `/usr/lib/systemd/user` instead.

File: data/wayland-proxy-virtwl.service (hunk 1 above)
Comment (Drakulix) [reply data/wayland-proxy-virtwl.service:7]: Also:
- Lets add `ExecStartPost=/usr/bin/systemctl --user set-environment DISPLAY=":0"`
- Also don't use `wayland-0`, that socket has a special fallback meaning. Rather use something custom, like `wayland-virtwl`
- Additionally we can also add and install section:
```
[Install]
WantedBy = cosmic-session.target
```

**Review by Drakulix — COMMENTED** [review]
A few remarks on the systemd-service file. Otherwise this looks good! thanks :)

File: data/wayland-proxy-virtwl.service (hunk 1 above)
Comment (wash2) [reply data/wayland-proxy-virtwl.service:7]: ok thanks, i'll make the changes. I saw the dpi & unscaling as suggestions from the README, but ya users can probably override it themselves.

Comment (wash2) [reply data/wayland-proxy-virtwl.service:7]: And I was setting $DISPLAY in start-cosmic, but i'll do that here as you've recommended instead.

File: justfile (hunk 2 above)
Comment (wash2) [reply justfile:49]: it seems /usr/lib does not work yet in ubuntu

**Pull request merged**